util/dvsim/results_server.py

```diff
@@ -0,0 +1,123 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Code for a wrapper class which represents the "results server".
+
+This is hosted with Google cloud.
+"""
+
+import datetime
+import logging as log
+import subprocess
+from shutil import which
+from typing import List, Optional
+
+
+class NoGCPError(Exception):
+    """Exception to represent "GCP tools are not installed"."""
+
+    pass
+
+
+class ResultsServer:
+    """A class representing connections to GCP (the results server)."""
+
+    def __init__(self, bucket_name: str):
+        """Construct results server; check gsutil is accessible."""
+        self.bucket_name = bucket_name
+
+        if which('gsutil') is None or which('gcloud') is None:
```

Comment:
It may be worth adding a comment mentioning that this assumes the user can authenticate for the project/bucket chosen, plus appropriate read/write permissions are set for that bucket.
In the future it would be nice to check this is true in the constructor.